variables.tf

```diff
@@ -154,6 +155,23 @@ variable "private_ips" {
   default     = []
 }
 
+variable "public_ip" {
+  description = "OCID of the Public IP to attach to primary vnic. Valid values are NONE, RESERVED or EPHEMERAL."
+  type        = string
+  default     = "NONE"
+
+  validation {
+    condition     = contains(["NONE", "RESERVED", "EPHEMERAL"], var.public_ip)
```

Comment:
Neat. I'll steal this :)

Reply:
I will backport similar behavior to vcn for the NAT Gateway reserved IP.